packages/common/src/store/pages/chat/slice.ts

```diff
@@ -140,6 +141,7 @@ const slice = createSlice({
       action: PayloadAction<{ unreadMessagesCount: number }>
     ) => {
       state.unreadMessagesCount = action.payload.unreadMessagesCount
+      state.optimisticUnreadMessagesCount = action.payload.unreadMessagesCount
```

Comment:
maybe don't do this actually... this will clear any pending optimistic changes. Perhaps instead of an "optimisticUnreadMessagesCount" we should keep track of "optimisticUnreadMessageCountChanges"?

Reply:
- UI shows 10 unreads
- start fetching unread count
- start mark chat as read that has 4 unreads
  - optimistic unread count decremented 4 to 6
  - UI shows 6 unreads
- fetch unread count succeeds with 10 since the write hasn't completed yet
  - optimistic unread count set to unread count, 10
  - UI shows 10 unreads
- mark chat as read succeeds
  - unread count updated to optimistic unread count (10)
  - optimistic unread count deleted
  - UI shows 10 unreads

Reply:
When should we reset `optimisticUnreadMessageCountChanges`? We need to reconcile it with the actual unread count eventually, right? Eg:
- On first load we fetch unread count, it's 4.
- We mark a chat as read that had all 4 unread messages, optimisticUnreadCountChanges = -4.
- 2 new messages come in on a different chat, optimisticUnreadCountChanges = -2
- Open left nav bar which re-fetches unreadCount, it is now 2.
- We would incorrectly show no unread messages bc unreadCount + optimisticUnreadCountChanges = 0.

Reply:
Hmm yeah this is quite tricky... Before I say anything, I think we could `delete state.optimisticUnreadMessagesCount` here and merge this as v1 and revisit before launch

Wonder if we could get away with only fetching the unread message count once, maybe on initial app load. The websocket is always listening, right? It'll "reconcile" on app kill + reopen. It sounds a little risky but it might not be...

Might have to sleep on this and think more